### ISSUE TITLE
 add more than one subtitle to the video

### ISSUE THREAD

**Issue by rasool-g** [ZeBobo5/Vlc.DotNet#432]
I have an issue / a question about Vlc.DotNet.

# Generic information

- **Vlc.DotNet version** : (2.2.1)
- **Vlc.DotNet project used** : (WinForms)
- **libvlc version** : (3.0.3) (x86/x64) 
- **.net version** : (4.5.2)
- **Project language** : (C#)
- **Project build architecture** : (x86/x64)
- **Operating system** : (Windows 10) (x64)

# Summary

Hello,
I need to add more than one subtitle to the video. But I can only add one subtitle before starting the video by using the following code.

`string[] opts = {  @"sub-file=" + textBox1.Text };
 vlcControl1.SetMedia(new System.IO.FileInfo(videoDir), opts);`

In the vlc media player, there is the option of "add subtitle File..." which allows the user to add any number of subtitles to the video at any time that they wish to. How can I embed such a feature to my application?

- [ ] I confirm that my issue doesn't happen in VLC itself.

**Comment by jeremyVignelles** [comment]
I don't know how to do that. Is there an option to do that in libvlc?

**Comment by chiefmb** [comment]
I have same problem. but in wpf and vlc.dotnet 3.

**Comment by jeremyVignelles** [comment]
I'm sorry, but I don't think there is an option for that in libvlc. I guess you will need to ask for that feature in vlc forum or in vlc bugtracker

**Comment by chiefmb** [comment]
vlc player is opensource too. can't you check the source and find out how it works? i don't know c and vlc as much as you to do it myself.

**Comment by jeremyVignelles** [comment]
I'm not familiar with every feature of libvlc. The easiest way to get help for that is to ask libvlc devs directly and ask them questions. Unfortunately, I'm maintaining this project on my free time and I can't make the intermediate for every question you ask about generic libvlc usage.
The question to ask in the forum would be "Is it possible to do XXX in libvlc". If the answer is yes, then link the topic here and I can help you with the implementation.

When there is an easy way, there is a function in [the doc](https://www.videolan.org/developers/vlc/doc/doxygen/html/group__libvlc__media.html#gaa72710d446a889df8ee40a6c0c65465d)

Maybe this can be done with "input slaves", but I'm not sure that this API is in Vlc.DotNet, and I don't know how to use it.

**Comment by chiefmb** [comment]
can you give a hand and i will add this feature.

**Comment by jeremyVignelles** [comment]
Yes, you can join the discussion on gitter. However, I'm not very available until monday. Did you have an answer from videolan?

**Comment by chiefmb** [comment]
I think so. there is an function named "libvlc_media_slaves_add" [here](https://www.videolan.org/developers/vlc/doc/doxygen/html/group__libvlc__media.html#gab21524c105f6dd706dd4118e026403f7)
With this it's possible to add subtitle to media. but it have some warning I don't know how it effect on this.

This function must be called before the media is parsed (via libvlc_media_parse_with_options()) or before the media is played (via libvlc_media_player_play())

**Comment by jeremyVignelles** [comment]
This means you can't use Play() directly. you need to SetMedia, AddSlave, and only then call Parse or Play on the media object

**Comment by chiefmb** [comment]
So is it still possible adding subtitle while movie is playing?

**Comment by chiefmb** [comment]
can't we pause or some thing, add sub and replay it??

**Comment by jeremyVignelles** [comment]
No, you would need to parse the file again.
You can create a new media based on the already playing file, add the media slave and play the new media, seek to the point you were. you will probably experience horrible flickerings...

**Comment by chiefmb** [comment]
damn it. thanks any way.

**Comment by tigros** [comment]
i'm looking to add a subtitle facility to my app, so far the only thing that works is --sub-file=h:\movies\English test.srt the other things like --sub-autodetect-path and --sub-autodetect-fuzzy=1 don't seem to work. maybe just sub-file will be enough but it's strange the others don't work. For example there's sometimes a Subs sub folder with .srt, be nice to load them automatically and let user choose from list instead of open file dialog. it's lucky sub-file works tho that could sure come in handy! but like you said you have to redo a SetMedia and seek oh well.
